client/blockchain-service/src/handler.rs

```diff
@@ -1046,4 +1092,49 @@ impl BlockchainService {
         };
         current_tick == &last_tick_provided.saturating_add(provider_challenge_period)
     }
+
+    fn check_pending_forest_root_writes(&mut self) {
+        if let Some(mut rx) = self.forest_root_write_lock.take() {
+            // TODO: consider adding a timeout here to prevent a task from blocking the runtime.
+            match rx.try_recv() {
+                // If the channel is empty, means we still need to wait for the current task to finish.
+                Err(TryRecvError::Empty) => self.forest_root_write_lock = Some(rx),
+                _ => {}
+            }
+        }
+
+        // If we have a task writing to the runtime, we don't want to start another one.
+        if self.forest_root_write_lock.is_some() {
+            return;
+        }
+
+        let (tx, rx) = tokio::sync::oneshot::channel();
+        self.forest_root_write_lock = Some(rx);
+
```

Comment:
This whole flow is a bit confusing. Probably worth adding some comments that explain:
Case 1: there is a message through the one-shot channel => task that had the lock has finished.
Case 2: there are no messages in the receiver => task that has the lock is still using it.
Case 3: the lock is None from the beginning, meaning no task has it.

Or restructure the code in a way that this is more evident.